napari/layers/shapes/shapes.py

```diff
@@ -535,8 +541,25 @@ def data(self):
         return self._data_view.data
 
     @data.setter
-    def data(self, data, shape_type='rectangle'):
+    def data(self, data):
         self._finish_drawing()
+
+        data, shape_type = extract_shape_type(data)
+        n_new_shapes = number_of_shapes(data)
+        # not given a shape_type through data
+        if shape_type is None:
```

Comment:
Can we do this same trick with the `edge_widths`, `edge_color`, `face_color`, and `z_indices` too.